src/python/gudhi/reader_utils.cc

```diff
@@ -0,0 +1,108 @@
+
+#include <vector>
+#include <array>
+#include <string>
+
+namespace py = pybind11;
+
+using Persistence_intervals = std::vector<std::array<double,2>>;
+PYBIND11_MAKE_OPAQUE(Persistence_intervals);
+
+PYBIND11_MODULE(reader_utils, m) {
+      m.attr("__license__") = "MIT";
+      m.def("read_points_from_off_file", [](const std::string& off_file) {
+            Gudhi::Points_off_reader<std::vector<double>> off_reader(off_file);
+            return off_reader.get_point_cloud();
+          },
+        py::arg("off_file"),
+        R"pbdoc(
+    Read points from OFF file.
+
+    :param off_file: Path to an `OFF <fileformats.html#off-file-format>`_ file.
+    :type off_file: string
+
+    :returns:  The point set.
+    :rtype: List[List[float]]
+        )pbdoc"
+      );
+      m.def("read_lower_triangular_matrix_from_csv_file", &Gudhi::read_lower_triangular_matrix_from_csv_file<double>,
+        py::arg("csv_file"), py::arg("separator") = ';',
+        R"pbdoc(
+    Read lower triangular matrix from a CSV style file.
+
+    :param csv_file: Path to a CSV file.
+    :type csv_file: string
+    :param separator: The value separator in the CSV file. Default value is ';'
+    :type separator: char
+
+    :returns:  The lower triangular matrix.
+    :rtype: List[List[float]]
+        )pbdoc"
+      );
+      m.def("read_persistence_intervals_grouped_by_dimension", &Gudhi::read_persistence_intervals_grouped_by_dimension,
+        py::arg("persistence_file"),
+        R"pbdoc(
+    Reads a file containing persistence intervals.
+    Each line might contain 2, 3 or 4 values: [[field] dimension] birth death
+    The return value is a `dict(dim, list(tuple(birth, death)))`
+    where `dim` is an `int`, `birth` a `float`, and `death` a `float`.
+    Note: the function does not check that birth <= death.
+
+    :param persistence_file: Path to a `persistence diagram <fileformats.html#persistence-diagram>`_ file.
+    :type persistence_file: string
+
+    :returns:  The persistence pairs grouped by dimension.
+    :rtype: Dict[int, List[Tuple[float, float]]]
+        )pbdoc"
+      );
+      py::bind_vector<Persistence_intervals>(m, "PersistenceIntervals", py::buffer_protocol());
+      m.def("read_persistence_intervals_in_dimension", [](const std::string& persistence_file, int only_this_dim) {
+            py::gil_scoped_release release;
+            // std::vector<std::pair<double, double>> is not exposable - cf. vector_has_data_and_format in pybind11/stl_bind.h
+            std::vector<std::array<double,2>> ret;
+            Gudhi::read_persistence_intervals_and_dimension(persistence_file,
+              boost::make_function_output_iterator([only_this_dim, &ret](std::tuple<int, double, double> t) {
+                if (only_this_dim == get<0>(t) || only_this_dim == -1) ret.push_back({get<1>(t), get<2>(t)});
+              }));
+            py::gil_scoped_acquire acquire;
```

Comment:
The `scope` in the name means that their destructor does the opposite action. So you are not supposed to have both release and acquire, you are just supposed to add appropriate `{` and `}` to ensure that the release object is destructed here.

Reply:
Sorry about that, I completely missed how it was working. Done on cee6228c